inventory tags: improve outline and add configurable thickness

winterdaze opened the pull request:
closes #6844 
closes #9768

Improves inventory tag outlines and adds a config item to choose outline thickness
Before: 
![before](https://i.imgur.com/vxk6r2s.png)
After: 
![after](https://i.imgur.com/9XdgcVS.png)
After with thickness: 
![after2](https://i.imgur.com/aWBHw6Y.png)

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/game/ItemManager.java:
```diff
@@ -478,12 +480,13 @@ public AsyncBufferedImage getImage(int itemId, int quantity, boolean stackable)
 	 * @return image
 	 */
-	private BufferedImage loadItemOutline(final int itemId, final int itemQuantity, final Color outlineColor)
+	private BufferedImage loadItemOutline(final int itemId, final int itemQuantity, final Color outlineColor, final int thickness)
```
Maybe instead of adding a breaking change to this API, add an overload with the default being 1 thickness.

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/game/ItemManager.java:
```diff
@@ -478,12 +480,13 @@ public AsyncBufferedImage getImage(int itemId, int quantity, boolean stackable)
 	{
-		final SpritePixels itemSprite = client.createItemSprite(itemId, itemQuantity, 1, 0, 0, true, 710);
-		return itemSprite.toBufferedOutline(outlineColor);
+		final BufferedImage outline = ImageUtil.outlineImage(getImage(itemId, itemQuantity, false), outlineColor, false, thickness);
+		return outline;
```
```suggestion
		return ImageUtil.outlineImage(getImage(itemId, itemQuantity), outlineColor, false, thickness);
```

The default for `outlineCorners` is false already. This also does not need to be saved to a variable since it is simply returned on the next line.

winterdaze replied:
the `false` you removed is for `stackable`, not `outlineCorners`
stackable should be false so the quantity text doesn't get outlined as well

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/util/ImageUtil.java:
```diff
@@ -326,6 +326,22 @@ public static BufferedImage outlineImage(final BufferedImage image, final Color
+	 * @return               The BufferedImage with its edges--and optionally, corners--outlined
+	 *                       with the given color.
+	 */
+	public static BufferedImage outlineImage(final BufferedImage image, final Color color, final Boolean outlineCorners, final int thickness)
```
Another overload of signature `BufferedImage image, Color color, int thickness` can be added to reduce verbisoty in usage.

Broooklyn commented:
I've noticed the normal white outline from the `Use` option doesn't work with this enabled, i.e., there is no way to tell if you're using an item on another without checking the text in the top left corner or the tooltip.

![javaw_JpcxKV1jfL](https://user-images.githubusercontent.com/54762282/93154974-ff98f200-f6d2-11ea-9e50-38e1f0444d52.png)

winterdaze commented:
> I've noticed the normal white outline from the `Use` option doesn't work with this enabled, i.e., there is no way to tell if you're using an item on another without checking the text in the top left corner or the tooltip.
> 
> ![javaw_JpcxKV1jfL](https://user-images.githubusercontent.com/54762282/93154974-ff98f200-f6d2-11ea-9e50-38e1f0444d52.png)

it looks like a mess in the old version too. where is the outline behavior of 'use' defined?
![img](https://i.imgur.com/EackJ3d.png)

Broooklyn commented:
It seems like this is taking priority over any change to the inventory icons, not just the `Use` white outline. I've noticed that it also doesn't darken/fade the sprite when you click to equip, for example.

winterdaze commented:
fixed the issue with the overlay/outline taking priority over interactions. also, the white outline from 'use' shows more now, even with a thicker border. is it good enough?
Updated:
![updatedUse](https://i.imgur.com/mloSXlN.png)
Updated with thickness:
![updatedUseThicker](https://i.imgur.com/bNZuf1f.png)
Old:
![oldUse](https://i.imgur.com/5A9K5x2.png)
as you can see in the old version, you could only see like two pixels of the white 'use' outline

winterdaze commented:
updated the PR, it now fixes rendering text over the outline

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/inventorytags/InventoryTagsOverlay.java:
```diff
@@ -63,8 +64,29 @@ public void renderItemOverlay(Graphics2D graphics, int itemId, WidgetItem itemWi
+					if (itemManager.getItemComposition(itemId).isStackable())
+					{
+						graphics.setFont(FontManager.getRunescapeSmallFont());
+						String quantityText = Integer.toString(quantity);
```
This strikes me as something which should be added to QuantityFormatter's API instead of only being used locally here.

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/util/QuantityFormatter.java:
```diff
@@ -145,6 +145,29 @@ public static synchronized String quantityToRSDecimalStack(int quantity, boolean
+	 */
+	public static String intToRSQuantity(int quantity)
+	{
+		String quantityText = Integer.toString(quantity);
+		if (quantity >= 10000000)
+		{
+			quantityText = quantityText.substring(0, quantityText.length() - 6) + "M";
+
+		}
+		else if (quantity >= 100000)
+		{
+			quantityText = quantityText.substring(0, quantityText.length() - 3) + "K";
+		}
+		return quantityText;
```
Avoiding multiple string operations would be better here.
```java
if (quantity >= 10_000_000)
{
	return Integer.toString(quantity / 1_000_000) + 'M';
}
else if (quantity >= 100_000)
{
	return Integer.toString(quantity / 1_000) + 'K';
}

return Integer.toString(quantityText);
```

Nightfirecat commented:
I've rebased this and reordered the commits to make more sense. This looks fine to me, and does seem to be identical to ingame item text. :+1:
